### PR TITLE
Improve user selection for evaluations

### LINKED ISSUE #2516: Improve user selection in evaluations
For user selection, the material design spec favors using checkboxes instead of buttons for selection. Selected rows should also have a slight background color:
image
vs
image
In addition, clicking the add or remove button reloads the entire table which I think is not necessary. It also breaks the flow when adding or removing someone that is not on the first page.

_Originally posted by @bmesuere in https://github.com/dodona-edu/dodona/issues/2462#issuecomment-786515640_ (and extracted into another issue, since the PR is big enough already).

### PR DESCRIPTION
**Pull request by chvp**
This pull request changes the look of the evaluation user edit table. It also makes sure we only refresh necessary rows instead of refreshing the entire table on each change.

screenshot_2021-08-03-175318

Closes #2516.

### CODE REVIEW

File: app/views/evaluations/add_users.html.erb [hunk 1]
```diff
@@ -64,4 +64,5 @@
 </div>
 <script>
   $(window.dodona.interceptAddMultiUserClicks);
+  $(dodona.initCheckboxes)
```
Comment (bmesuere) [inline app/views/evaluations/add_users.html.erb:67]: Why is one prefixed with `window` and the other not?

File: config/locales/views/evaluations/nl.yml [hunk 2]
```diff
@@ -82,10 +82,9 @@ nl:
       invisible_total: "De totaalscore werd niet vrijgegeven"
       no_grading: "Deze oefening staat niet op punten"
     members_table:
-      remove_user: Student verwijderen uit evaluatie
-      remove_user_consequences: "Als deze student al feedback gekregen had, dan zal deze verwijderd worden. Ben je zeker dat je deze student wil verwijderen?"
       remove_users_consequences: "Als deze studenten al feedback gekregen hadden, dan zal deze verwijderd worden. Ben je zeker dat je deze studenten wil verwijderen?"
-      add_user: Student toevoegen aan evaluatie
       status: Voortgang
+    member_row:
+      remove_user_consequences: "Als deze student al feedback gekregen had, dan zal deze verwijderd worden. Ben je zeker dat je deze student wil verwijderen?"
```
Comment (bmesuere) [inline config/locales/views/evaluations/nl.yml:88]:
```suggestion
      remove_user_consequences: "Als deze student al feedback gekregen had, dan zal deze feedback verwijderd worden. Ben je zeker dat je deze student wil verwijderen?"
```

File: config/locales/views/evaluations/en.yml [hunk 3]
```diff
@@ -83,10 +83,9 @@ en:
       invisible_total: "The total score was not released"
       no_grading: "This exercise was not graded"
     members_table:
-      remove_user: Remove student from evaluation
-      remove_user_consequences: "If this student already received feedback, it will be deleted. Are you sure you want to remove this student?"
       remove_users_consequences: "If these students already received feedback, it will be deleted. Are you sure want to remove these students?"
-      add_user: Add student to evaluation
       status: Progress
+    member_row:
+      remove_user_consequences: "If this student already received feedback, it will be deleted. Are you sure you want to remove this student?"
```
Comment (bmesuere) [inline config/locales/views/evaluations/en.yml:89]:
```suggestion
      remove_user_consequences: "If this student already received feedback, that feedback will be deleted. Are you sure you want to remove this student?"
```

**Review by bmesuere — APPROVED** [review]

File: app/views/evaluations/add_users.html.erb (hunk 1 above)
Comment (chvp) [reply app/views/evaluations/add_users.html.erb]: I tried to be consistent, apparently forgot that here. Fixed.

**Review by niknetniko — COMMENTED** [review]
I noticed one small bug: when unchecking the checkbox, and pressing cancel on this dialog:
image

The checkbox is still unchecked, but the user is still included as expected; a refresh fixes things:
image

**Comment by chvp** [comment]
Should be fixed now.

**Review by niknetniko — APPROVED** [review]

**Pull request merged**